lib/toggle-view.js

```diff
@@ -0,0 +1,90 @@
+/* @flow */
+
+import ConfigFile from 'sb-config-file'
+import SelectListView from 'atom-select-list'
+import { CompositeDisposable, Emitter, Disposable } from 'atom'
+import { LINTER_CONFIG_FILE_PATH, LINTER_CONFIG_FILE_DEFAULT } from './helpers'
+
+type ToggleAction = 'enable' | 'disable'
+
+export default class ToggleProviders {
+  action: ToggleAction;
+  config: ConfigFile;
+  emitter: Emitter;
+  providers: Array<string>;
+  subscriptions: CompositeDisposable;
+
+  constructor(action: ToggleAction, providers: Array<string>) {
+    this.action = action
+    this.config = null
+    this.emitter = new Emitter()
+    this.providers = providers
+    this.subscriptions = new CompositeDisposable()
+
+    this.subscriptions.add(this.emitter)
+  }
+  async getConfig(): Promise<ConfigFile> {
+    if (!this.config) {
+      this.config = await ConfigFile.get(LINTER_CONFIG_FILE_PATH, LINTER_CONFIG_FILE_DEFAULT, {
+        prettyPrint: true,
+        createIfNonExistent: false,
+      })
+    }
+    return this.config
+  }
+  async getItems(): Promise<Array<string>> {
+    const disabled = await (await this.getConfig()).get('disabled')
+    if (this.action === 'disable') {
+      return this.providers.filter(name => disabled.indexOf(name) === -1)
```

Comment:
You could just do `disabled.includes(name)` instead of the older `disabled.indexOf(name) === -1`, leads to clearer code.